### ISSUE TITLE
False unexpected-keyword-arg for classes defined differently in branches.

### ISSUE THREAD

**Issue by kayhayen** [pylint-dev/pylint#3603]
### Steps to reproduce

```python
if str is bytes:

   class C(object):
      def __init__(self, a):
            pass
else:
   class C(object):
      def __init__(self, a, b):
            pass
 
C(1, b=2)
```

In Nuitka, this is a pattern to hide e.g. `str` builtin differences, and I consider it valid code.

### Current behavior

pylint_problem.py:10 E1123 unexpected-keyword-arg  Unexpected keyword argument 'b' in constructor call

### Expected behavior

Since pylint cannot know, it should accept either signature, or shut up.

### pylint --version output
pylint 2.5.2
astroid 2.4.1
Python 3.7.3 (default, Dec 20 2019, 18:57:59) 
[GCC 8.3.0]
